router/http/src/main/scala/io/buoyant/router/http/DiscardProxySideEffects.scala

```diff
@@ -0,0 +1,31 @@
+package io.buoyant.router.http
+
+import com.twitter.finagle.{Service, ServiceFactory, SimpleFilter, Stack}
+import com.twitter.finagle.http.{Request, Response}
+
+object DiscardProxySideEffects {
+
+  /**
+   * Strips host from request URI and drops Proxy-* headers
+   * Those side-effects are added by proxy-aware clients
+   */
+  object filter extends SimpleFilter[Request, Response] {
+    def apply(req: Request, svc: Service[Request, Response]) = {
+      if (!req.uri.startsWith("/")) { // Proxied request
+        // split URI like http://acme.co/foo/bar?x=1 into host 'acme.co' and uri 'foo/bar?x=1'
+        val Array(_, _, host, uri) = req.uri.split("/", 4)
```

Comment:
this will throw an exception if there aren't 4 elements returned.  we should be more tolerant of unexpected input:

```
req.uri.split("/", 4) match {
  case Array(...) =>
  case _ => // doesn't match ...
}